harmonizer/js/do_compose.js

```diff
@@ -0,0 +1,44 @@
+/** @typedef {{typeDefs: string, name: string, url?: string;}} ServiceDefinition */
+
+/**
+ * @type {ServiceDefinition[]}
+ */
+var serviceList = serviceList;
+
+if (!serviceList || !Array.isArray(serviceList)) {
+  throw new Error("Error in JSRustland: serviceList missing or incorrect.");
+}
+
+serviceList.some((service) => {
+  if (
+    typeof service.name !== "string" || !service.name ||
+    typeof service.url !== "string" && service.url ||
+    typeof service.typeDefs !== "string" && service.typeDefs
+  ) {
+    throw new Error("Missing required data structure on service.");
+  }
+});
+
+
+serviceList = serviceList.map(({ typeDefs, ...rest }) => ({
+  typeDefs: parseTypedefs(typeDefs),
+  ...rest,
+}));
+
+function parseTypedefs(source) {
+  try {
+    return composition.parseGraphqlDocument(source)
+  } catch (err) {
+    print(err.message)
```

Comment:
Why are we printing these errors, but throwing other errors ourselves?

Reply:
Good catch! Applied 3f65f58 which will invoke `done({ Err: err })`.

cc @queerviolet who introduced this in what I have reason to believe was usefulness in debugging this during build-out.  Perhaps not necessary now, but a test could help us be more certain?

Reply:
Yeah, I was trying to debug what turned out to be an error in parsing selection sets. Our error reporting in that case is miserable, but it appears to just be miserable throughout the typescript implementation, so we can fix it there later.

I did apply e93a308, which try/catches composition errors and reports them via the `Result` mechanism, rather than panicking.

Reply:
Nice.  Thanks for the patch / follow-up, @queerviolet.